### PR TITLE
Add typings for Course root pages

### PR DESCRIPTION
**Pull request by ITsolution-git**
## Description
Add Types for firebase data structure and component's proptypes

## Affected Dependencies
None

## How has this been tested?
None

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: apps/courses/src/components/Header.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ import {
   faCog,
   faCommentAlt,
 } from '@fortawesome/free-solid-svg-icons';
-import { User } from '@openmined/shared/types';
+import { OpenMinded } from '@openmined/shared/types';
```
Comment (cereallarceny) [inline apps/courses/src/components/Header.tsx]: Hahaha our name is "OpenMined", not "OpenMinded". That's Open + Mined (to mine something in the past tense), not to be confused with "Mind" or "Mine".

File: libs/shared/types/src/index.ts [hunk 2]
```diff
@@ -1,22 +1,103 @@
...
+  }
+
+  export type LessonConceptQuiz = {
+    correct: number;
+    percentage: number;
+    total: number;
+  }
+
+  export type Project = {
+    started_at: firebase.firestore.DocumentReference;
```
Comment (cereallarceny) [inline libs/shared/types/src/index.ts]: Should be `firebase.firestore.Timestamp`

File: libs/shared/types/src/index.ts [hunk 3]
```diff
@@ -1,22 +1,103 @@
...
+    completed_at?: firebase.firestore.DocumentReference;
```
Comment (cereallarceny) [inline libs/shared/types/src/index.ts]: Same here

**Review by cereallarceny — CHANGES_REQUESTED** [review]

File: apps/courses/src/components/Header.tsx (hunk 1 above)
Comment (ITsolution-git) [reply apps/courses/src/components/Header.tsx]: Urgg, sorry I am a dumb. (that's why I wear glasses...) I thought it was open-minded - empathetic!!

**Review by cereallarceny — APPROVED** [review]

**Pull request merged**